mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -41,35 +41,74 @@ const createUserPreviewWithUpdatedFavorites = (user: UserPreview, updatedFavorit
 };
 
 /**
- * Delete favoriteWayUuid from favoriteWay array
+ * Create way with updated favorites
  */
-const deleteFavoriteWayUuid = async (
+const createWayWithUpdatedFavorites = (way: WayPreview, updatedFavoriteForUserUuids: string[]) => {
+  return new WayPreview({
+    uuid: way.uuid,
+    name: way.name,
+    dayReportUuids: way.dayReportUuids,
+    owner: way.owner,
+    goal: way.goal,
+    mentors: way.mentors,
+    mentorRequests: way.mentorRequests,
+    isCompleted: way.isCompleted,
+    lastUpdate: way.lastUpdate,
+    favoriteForUserUuids: updatedFavoriteForUserUuids,
+    createdAt: way.createdAt,
+    wayTags: way.wayTags,
+    jobTags: way.jobTags,
+  });
+};
+
+/**
+ * Add way uuid to UserPreview favoriteWays and add user uuid to WayPreview favoriteForUserUuids
+ */
+const addFavoriteToWayAndToUser = async (
   userPreview: UserPreview,
-  wayUuid: string,
+  wayPreview: WayPreview,
   setUser: React.Dispatch<React.SetStateAction<UserPreview | null>>,
+  setWay: React.Dispatch<React.SetStateAction<WayPreview | undefined>>,
 ) => {
```

Comment:
Is way could be undefined? I think better to use null )

Reply:
No way.